### ISSUE TITLE
sync of local user deletes the local user

### ISSUE THREAD

**Issue by phil-davis** [owncloud/user_ldap#511]
```
  Scenario: sync a local user
    Given user "local-user" has been created with default attributes in the database user backend
    When the administrator changes the display name of user "local-user" to "Test User" using the occ command
    And LDAP user "local-user" is resynced
    Then the command should have been successful
    And user "local-user" should not exist
```

This scenario looks a bit surprising to me. The test does the command:
```
occ user:sync OCA\User_LDAP\User_Proxy -u local-user -m remove
```

`local-user` is in the local user database, nit in LDAP. So of course it is not found in LDAP. The admin has specified `-m remove` so the command removes the user. But the user is (and always was) just in the local database, they never came from LDAP in the first place.

If I was the admin then I would be a bit surprised that the `user:sync` command could ever delete a "real local user", even when I use `-m remove`. I feel like I should get an error message whenever the UID  specified after `-u` is a "real local user". I should not be able to accidentally mess up a "real local user" with the `user:sync`  command. I should have to explicitly do a `user:delete` if I want to delete a "real local user" (e.g. before then doing `user:sync` to sync the same UID that has just been added to LDAP)

**Comment by phil-davis** [comment]
@pmaier1 @micbar or anybody - opinion?

**Comment by pmaier1** [comment]
Yes, I fully agree. Great spot @phil-davis. The command's use case is to clean-up **LDAP users** which are not available **in LDAP any more**. The command should not delete users that do not come from LDAP.

**Comment by phil-davis** [comment]
Added the "bug" label.
The fix is likely to be in core code (just a guess, because the `user:sync` command is in core)
@micbar please assign priority... for dev action.

**Comment by phil-davis** [comment]
I took myself off assignment, this now needs to be scheduled for developer work...
CC @micbar
